### PR TITLE
chore: migrate E2E deployment test to Foundry

### LINKED ISSUE #149: tests(e2e): `deployment`

### PR DESCRIPTION
**Pull request by meetmangukiya**
## Description

Migrate the deployment e2e test to foundry. Depends on https://github.com/cowprotocol/contracts/pull/215.

## Test Plan

CI

## Related Issues

Closes #149

### CODE REVIEW

File: test/e2e/Deployment.t.sol [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    function test__determininstic_addresses__settlement() external view {
+        assertEq(
+            _computeCreate2Addr(
+                abi.encodePacked(vm.getCode("GPv2Settlement"), abi.encode(address(authenticator), address(vault)))
+            ),
+            address(settlement),
+            "settlement address not as expected"
+        );
+    }
```
Comment (meetmangukiya) [inline test/e2e/Deployment.t.sol:60]: these are failing, not very apparent why, will circle back later.

Comment (meetmangukiya) [reply test/e2e/Deployment.t.sol:60]: got it fixed: e60d1c8

File: test/e2e/Deployment.t.sol [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        emit Metadata(hint, metadata);
+    }
+
+    function _computeCreate2Addr(bytes memory initCode) internal view returns (address) {
+        return vm.computeCreate2Address(SALT, hashInitCode(initCode), deployer);
+    }
+
+    function _implementationAddress(address proxy) internal view returns (address) {
+        return address(
+            uint160(uint256(vm.load(proxy, 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc)))
```
Comment (mfw78) [inline test/e2e/Deployment.t.sol:106]: nit: for a constant such as this, please declare at the top of the file and label how the constant is derived.

File: test/e2e/Deployment.t.sol [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        assertEq(
+            _computeCreate2Addr(
+                abi.encodePacked(vm.getCode("GPv2Settlement"), abi.encode(address(authenticator), address(vault)))
+            ),
+            address(settlement),
+            "settlement address not as expected"
+        );
+    }
+
+    function test__authorization__authenticator_hash_dedicated_owner() external view {
```
Comment (mfw78) [inline test/e2e/Deployment.t.sol:62]:
```suggestion
    function test__authorization__authenticator_has_dedicated_owner() external view {
```

File: test/e2e/Deployment.t.sol [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        assertEq(IEIP173Proxy(address(allowList)).owner(), owner, "owner not as expected");
+    }
+
+    function test__authorization__authenticator_hash_dedicated_manager() external view {
```
Comment (mfw78) [inline test/e2e/Deployment.t.sol:66]:
```suggestion
    function test__authorization__authenticator_has_dedicated_manager() external view {
```

**Review by mfw78 — APPROVED** [review]

**Comment by mfw78** [comment]
Merging without CI/CD green due secret for RPC.

**Pull request merged**